pkg/util/gc_service.go

```diff
@@ -30,10 +33,21 @@ const (
 
 // CheckSafetyOfStartTs checks if the startTs less than the minimum of Service-GC-Ts
 // and this function will update the service GC to startTs
-func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) error {
-	minServiceGCTs, err := pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
-		cdcChangefeedCreatingServiceGCSafePointTTL, startTs)
-	if err != nil {
+func CheckSafetyOfStartTs(ctx context.Context, pdCli pd.Client, changefeedID string, startTs uint64) (err error) {
+	var minServiceGCTs uint64
+	// pd leader switch may happen, so just retry it.
+	if err := retry.Do(ctx, func() error {
+		minServiceGCTs, err = pdCli.UpdateServiceGCSafePoint(ctx, cdcChangefeedCreatingServiceGCSafePointID+changefeedID,
+			cdcChangefeedCreatingServiceGCSafePointTTL, startTs)
+		if err != nil {
+			log.Warn("update GC safepoint failed, retry later", zap.Error(err))
+		}
+		return err
+	},
+		retry.WithBackoffBaseDelay(500),
+		retry.WithBackoffMaxDelay(60*1000),
```

Comment:
Is `60*1000` used as the max delay latency of one time retry, if the max retry time is 8, the retry latency of last time is 0.5 * (1.5 ^ 8) = 25.6s, which is less than 60s, so maybe the option `retry.WithBackoffMaxDelay(60*1000)` is useless.